jlm/llvm/opt/alias-analyses/AgnosticMemoryNodeProvider.cpp

```diff
@@ -88,6 +90,48 @@ class AgnosticMemoryNodeProvisioning final : public MemoryNodeProvisioning
   }
 
 private:
+  [[nodiscard]] bool
+  GetOutputNodesFromRegisterNode(
+      const rvsdg::output & output,
+      util::HashSet<const PointsToGraph::MemoryNode *> & memoryNodes) const
+  {
+    const PointsToGraph::RegisterNode * registerNode;
+    try
+    {
+      registerNode = &PointsToGraph_.GetRegisterNode(output);
+    }
+    catch (...)
+    {
+      return false;
+    }
+
+    for (auto & memoryNode : registerNode->Targets())
+      memoryNodes.Insert(&memoryNode);
+
+    return true;
+  }
+
+  [[nodiscard]] bool
+  GetOutputNodesFromRegisterSetNode(
+      const rvsdg::output & output,
+      util::HashSet<const PointsToGraph::MemoryNode *> & memoryNodes) const
+  {
+    const PointsToGraph::RegisterSetNode * registerSetNode;
+    try
+    {
+      registerSetNode = &PointsToGraph_.GetRegisterSetNode(output);
```

Comment:
So the try-catch in this case is temporary and will disappear again once everything is ported to RegisterSetNodes.

Since we do not catch exceptions and handle them properly, it makes effectively no difference whether we throw an exception or use a `JLM_ASSERT` in the `Get*Node()` methods of the points-to graph. We should change this to asserts I guess, as it is a generally a bug when requesting a points-to graph node that does not exist. This should happen in another PR though.